### PR TITLE
Feat : Toggle Eye Icon to Expand/Collapse Kanban Card

### PR DESCRIPTION
**Pull request by Vardhaman619**
This pull request implements the functionality to toggle the eye icon in Kanban cards to expand or collapse the card details.

#7389 

[toogle-button-in-kanban-card.webm](https://github.com/user-attachments/assets/3bc1a31c-4053-429a-95e7-aa98188c39e4)

### CODE REVIEW

File: packages/twenty-front/src/modules/object-record/record-board/record-board-card/components/RecordBoardCard.tsx [hunk 1]
```diff
@@ -205,11 +205,7 @@ export const RecordBoardCard = ({
     </StyledFieldContainer>
   );
 
-  const onMouseLeaveBoard = () => {
-    if (isCompactModeActive) {
-      setIsCardInCompactMode(true);
-    }
-  };
+
```
Comment (ehconitin) [inline packages/twenty-front/src/modules/object-record/record-board/record-board-card/components/RecordBoardCard.tsx:208]: lets not remove onMouseLeaveBoard

Comment (ehconitin) [inline packages/twenty-front/src/modules/object-record/record-board/record-board-card/components/RecordBoardCard.tsx:208]:
```suggestion
    const onMouseLeaveBoard = () => {
    if (isCompactModeActive) {
      setIsCardInCompactMode(true);
    }
  };
```

File: packages/twenty-front/src/modules/object-record/record-board/record-board-card/components/RecordBoardCard.tsx [hunk 2]
```diff
@@ -243,7 +239,6 @@ export const RecordBoardCard = ({
       <StyledBoardCard
         ref={cardRef}
         selected={isCurrentCardSelected}
```
Comment (ehconitin) [inline packages/twenty-front/src/modules/object-record/record-board/record-board-card/components/RecordBoardCard.tsx:241]:
```suggestion
        selected={isCurrentCardSelected}
        onMouseLeave={onMouseLeaveBoard}
```

File: packages/twenty-front/src/modules/object-record/record-board/record-board-card/components/RecordBoardCard.tsx [hunk 3]
```diff
@@ -293,7 +288,7 @@ export const RecordBoardCard = ({
                     accent="tertiary"
                     onClick={(e) => {
                       e.stopPropagation();
-                      setIsCardInCompactMode(false);
+                      setIsCardInCompactMode((prev) => !prev);
                     }}
                   />
```
Comment (ehconitin) [inline packages/twenty-front/src/modules/object-record/record-board/record-board-card/components/RecordBoardCard.tsx:293]:
```suggestion
                     <LightIconButton
                    Icon={isCardInCompactMode ? IconEye : IconEyeOff}
                    accent="tertiary"
                    onClick={(e) => {
                      e.stopPropagation();
                      setIsCardInCompactMode((prev) => !prev);
                    }}
                  />
```

File: packages/twenty-front/src/modules/object-record/record-board/record-board-card/components/RecordBoardCard.tsx [hunk 4]
```diff
@@ -293,7 +288,7 @@ export const RecordBoardCard = ({
                     accent="tertiary"
```
Comment (ehconitin) [inline packages/twenty-front/src/modules/object-record/record-board/record-board-card/components/RecordBoardCard.tsx:288]: icon should be conditionally rendered
eg - 
   Icon={isCardInCompactMode ? IconEye : IconEyeOff}

**Review by ehconitin — CHANGES_REQUESTED** [review]
Left some comments! ;)

**Comment by Vardhaman619** [comment]
@ehconitin

**Comment by ehconitin** [comment]
@Bonapara 
Kept onMouseLeaveBoard with a debounced call of 800ms
Behaviour - 

https://github.com/user-attachments/assets/8945d2b0-9da0-4c11-9edb-f5c04ab23128

**Comment by ehconitin** [comment]
@FelixMalfait 
renamed isCardInCompactMode to isCardExpanded because its more of a toggle now.
made onMouseLeaveBoard to have a debounced call - can remove if card is to immediately close on mouse leave.
Thanks :)

**Comment by FelixMalfait** [comment]
Great points @ehconitin agree with both! Thank you

Thanks @Vardhaman619!

**Review by FelixMalfait — APPROVED** [review]

**Review by ehconitin — APPROVED** [review]

**Pull request merged**